### PR TITLE
Add status as part of DecoratorController hook response

### PR DESCRIPTION
**Pull request by lionelvillard**
In order to be consistent with CompositeController the status returned by the hook replaces the existing one

### CODE REVIEW

**Comment by lionelvillard** [comment]
@enisoc it should be good now.

File: controller/decorator/controller.go [hunk 1]
```diff
@@ -466,14 +467,18 @@ func (c *decoratorController) syncParentObject(parent *unstructured.Unstructured
...
 	annotationsChanged := updateStringMap(parentAnnotations, syncResult.Annotations)
+	statusChanged := !reflect.DeepEqual(parentStatus, syncResult.Status)
 
 	// Only do the update if something changed.
-	if labelsChanged || annotationsChanged ||
+	if labelsChanged || annotationsChanged || statusChanged ||
 		(syncResult.Finalized && dynamicobject.HasFinalizer(parent, c.finalizer.Name)) {
 		updatedParent.SetLabels(parentLabels)
 		updatedParent.SetAnnotations(parentAnnotations)
+		k8s.SetNestedField(updatedParent.Object, syncResult.Status, "status")
```
[the remark this answers is not in the document]
Comment (lionelvillard) [reply controller/decorator/controller.go]: There is a typo: `result, err := ...`. Not sure if I can edit it directly in github.

[the remark this answers is not in the document]
Comment (enisoc) [reply controller/decorator/controller.go]: Ah good catch. I'll fix that in a new suggested change. I think you just did a code review of my code review... This is a strange new world we live in.

```suggestion
		k8s.SetNestedField(updatedParent.Object, syncResult.Status, "status")

		if statusChanged && parentClient.HasSubresource("status") {
			// The regular Update below will ignore changes to .status so we do it separately.
			result, err := parentClient.Namespace(parent.GetNamespace()).UpdateStatus(updatedParent)
			if err != nil {
				return fmt.Errorf("can't update status: %v", err)
			}
			// The Update below needs to use the latest ResourceVersion.
			updatedParent.SetResourceVersion(result.GetResourceVersion())
		}
```

File: controller/decorator/controller.go [hunk 2]
```diff
@@ -466,14 +467,32 @@ func (c *decoratorController) syncParentObject(parent *unstructured.Unstructured
...
+
+		if statusChanged && parentClient.HasSubresource("status") {
+			// The regular Update below will ignore changes to .status so we do it separately.
+			result, err = parentClient.Namespace(parent.GetNamespace()).UpdateStatus(updatedParent)
```
Comment (enisoc) [inline controller/decorator/controller.go]:
```suggestion
			result, err := parentClient.Namespace(parent.GetNamespace()).UpdateStatus(updatedParent)
```

**Pull request merged**